pkg/controller/mongodb/mongodb_controller.go

```diff
@@ -220,10 +228,28 @@ func (r *ReplicaSetReconciler) isStatefulSetReady(mdb mdbv1.MongoDB, existingSta
 		return false, err
 	}
 
-	// comparison is done with bytes instead of reflect.DeepEqual as there are
-	// some issues with nil/empty maps not being compared correctly otherwise
+	//comparison is done with bytes instead of reflect.DeepEqual as there are
+	//some issues with nil/empty maps not being compared correctly otherwise
 	areEqual := bytes.Compare(stsCopyBytes, stsBytes) == 0
```

Comment:
While setting up the linter, I see it complaining about this, we should use

`bytes.Equal(stsCopyBytes, stsBytes)`

Reply:
(Of course it's not related to this PR so feel free to ignore and I'll do it on my linter PR!)